ksqldb-engine/src/main/java/io/confluent/ksql/analyzer/RewrittenAnalysis.java

```diff
@@ -113,7 +123,49 @@ public ImmutableAnalysis original() {
 
   @Override
   public Optional<WindowExpression> getWindowExpression() {
-    return original.getWindowExpression();
+    if (original.getRefinementInfo().isPresent()
```

Comment:
nit: I think this code would be easier to read if we had:
```java
if (!original.getRefinementInfo().isPresent || !original.getWindowExpression().isPresent()) {
  return original.getWindowExpression();
}

RefinementInfo refinement = optional.get();
WindowExpression window = optional.get();
...
```

Reply:
We should still check if there is a grace period though right, because if there is we also just return the original window expression

Reply:
yes we do need to check that

Reply:
So I rewrote the condition, but in order to pass the current tests I needed to add some conditions and I feel that it may be even harder to read now, my current if condition being
 `    if (!original.getRefinementInfo().isPresent() || !original.getWindowExpression().isPresent()
        || original.getWindowExpression().get().getKsqlWindowExpression()
        .getGracePeriod().isPresent()
        || original.getRefinementInfo().get().getOutputRefinement() == OutputRefinement.CHANGES
    ) {
      return original.getWindowExpression();
    }`